engines/suppliers/app/models/suppliers/suppliers_finder.rb

```diff
@@ -0,0 +1,30 @@
+module Suppliers
+  class SuppliersFinder
+
+    def initialize(user, group_id = nil)
+      @group_id = group_id
+      @user = user
+    end
+
+    def find
+      if group_id
```

Comment:
yes. `attr_reader` is just a shorthand for
```
def group_id
  @group_id
end
```
likewise `attr_writer` and `attr_accessor`.